lindera-ipadic-builder/src/error.rs

```diff
@@ -0,0 +1,50 @@
+use std::fmt;
+
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Clone, Copy, Eq, PartialEq, Serialize, Deserialize)]
+pub enum BuildErrorKind {
+    Content,
+    Decode,
+    Io,
+    Parse,
+    Serialize,
+}
+
+impl BuildErrorKind {
+    pub fn with_error<E>(self, source: E) -> BuildError
+    where
+        anyhow::Error: From<E>,
+    {
+        BuildError {
+            kind: self,
+            source: From::from(source),
+        }
+    }
+}
+
+#[derive(thiserror::Error, Debug)]
+#[error("BuildError(kind={kind:?}, source={source})")]
+pub struct BuildError {
```

Comment:
Could be better to make it more specific?
```suggestion
pub struct DictionaryBuildError {
```

Reply:
Thanks for the refactor!